test/tbb/test_parallel_phase.cpp

```diff
@@ -0,0 +1,288 @@
+                [this] { arena->start_parallel_phase(); },
+                [this] { arena->end_parallel_phase(with_fast_leave); });
+        } else {
+            median_start_time = measure_median_start_time(arena,
+                [] { tbb::this_task_arena::start_parallel_phase(); },
+                [this] { tbb::this_task_arena::end_parallel_phase(with_fast_leave); });
+        }
+        return median_start_time;
+    };
+
+public:
+    start_time_collection_sequenced_phases(tbb::task_arena& ta, std::size_t ntrials, bool fast_leave = false) :
+        base(ta, ntrials), with_fast_leave(fast_leave)
+    {}
+
+    explicit start_time_collection_sequenced_phases(std::size_t ntrials, bool fast_leave = false) :
+        base(ntrials), with_fast_leave(fast_leave)
+    {}
+};
+
+class start_time_collection_sequenced_scoped_phases
+    : public start_time_collection_base<start_time_collection_sequenced_scoped_phases>
+{
+  using base = start_time_collection_base<start_time_collection_sequenced_scoped_phases>;
+  friend base;
+
+  bool with_fast_leave;
+
+  std::size_t measure_impl() {
+      tbb::task_arena::scoped_parallel_phase* phase = nullptr;
+      auto median_start_time = measure_median_start_time(arena,
+          [this, &phase] {
+              phase = new tbb::task_arena::scoped_parallel_phase{*arena, with_fast_leave};
+          },
+          [&phase] {
+              delete phase;
+      });
+    return median_start_time;
+  };
+
+public:
+  start_time_collection_sequenced_scoped_phases(tbb::task_arena& ta, std::size_t ntrials, bool fast_leave = false) :
+      base(ta, ntrials), with_fast_leave(fast_leave)
+  {}
+
+  explicit start_time_collection_sequenced_scoped_phases(std::size_t ntrials, bool fast_leave = false) :
+      base(ntrials), with_fast_leave(fast_leave)
+  {}
+};
+
+//! \brief \ref interface \ref requirement
+TEST_CASE("Check that workers leave faster with leave_policy::fast") {
+    // Test measures workers start time, so no there is no point to
+    // measure it with workerless arena
+    if (utils::get_platform_max_threads() < 2) {
+        return;
+    }
+    tbb::task_arena ta_automatic_leave {
+        tbb::task_arena::automatic, 1,
+        tbb::task_arena::priority::normal,
+        tbb::task_arena::leave_policy::automatic
+    };
+    tbb::task_arena ta_fast_leave { 
+        tbb::task_arena::automatic, 1,
+        tbb::task_arena::priority::normal,
+        tbb::task_arena::leave_policy::fast
+    };
+    start_time_collection st_collector1{ta_automatic_leave, /*num_trials=*/10};
+    start_time_collection st_collector2{ta_fast_leave, /*num_trials=*/10};
+
+    auto times_automatic = st_collector1.measure();
+    auto times_fast = st_collector2.measure();
+
+    auto median_automatic = utils::median(times_automatic.begin(), times_automatic.end());
+    auto median_fast = utils::median(times_fast.begin(), times_fast.end());
+
+    WARN_MESSAGE(median_automatic < median_fast,
```

Comment:
Are all of the tests guaranteed to work (i.e. not have warnings) for hybrid systems where automatic is not delayed leave?

Reply:
That is a good question. I suppose that only one test case(Parallel Phase retains workers in task_arena) would produce expected results one hybrid systems. Other test don't make much sense on such configurations, but I think it is fine since these are warnings, not asserts.

Reply:
Can we skip the warning on hybrid systems?  For example, don't output warning if there is more than 1 core type.